src/plugins/intel_gpu/src/plugin/transformations/op/kv_cache.cpp

```diff
@@ -120,6 +120,17 @@ std::vector<ov::PartialShape> shape_infer(const KVCache* op, const std::vector<o
         out_shapes[0][concat_axis] += input_shapes[1][concat_axis];
     }
 
+    // Sometime input0 shape has zeros (or even dynamic dim) in several dimensions, for
+    // example concat [-1, 0, 0, 0] + [-1, 4, -1, 128] along axis 2, we could (and should) infer
+    // dim value of axis 1 and 4 in this case.
+    for (int64_t i = 0; i < static_cast<int64_t>(out_shapes[0].size()); ++i) {
```

Comment:
I think we can just initialize out_shapes[0] with input_shapes[1] instead of input_shapes[0], since the input_shapes[1] input shape is always "new_token" input and input_shapes[0] is "past". It seems [1] shape should always be more detailed

Reply:
done